samza-core/src/main/java/org/apache/samza/container/grouper/stream/AllSspToSingleTaskGrouperFactory.java

```diff
@@ -19,51 +19,83 @@
+ * consumed by each TaskInstance. This is useful, in case of using load-balanced consumers like the high-level Kafka
+ * consumer and Kinesis consumer, where Samza doesn't control the partitions being consumed by the task.
  *
- * Note: This grouper does not take in broadcast streams yet.
+ * Note:
+ * 1. This grouper does not take in broadcast streams yet.
+ * 2. This grouper is supported only with Yarn, Zk-standalone and Passthrough JobCoordinators.
  */
+
 class AllSspToSingleTaskGrouper implements SystemStreamPartitionGrouper {
-  private final int containerId;
+  private final Set<String> processorNames;
 
-  public AllSspToSingleTaskGrouper(int containerId) {
-    this.containerId = containerId;
+  AllSspToSingleTaskGrouper(Set<String> processorNames) {
+    this.processorNames = processorNames;
   }
 
   @Override
   public Map<TaskName, Set<SystemStreamPartition>> group(final Set<SystemStreamPartition> ssps) {
+    Map<TaskName, Set<SystemStreamPartition>> groupedMap = new HashMap<>();
+
     if (ssps == null) {
       throw new SamzaException("ssp set cannot be null!");
     }
     if (ssps.size() == 0) {
       throw new SamzaException("Cannot process stream task with no input system stream partitions");
     }
 
-    final TaskName taskName = new TaskName(String.format("Task-%s", String.valueOf(containerId)));
+    processorNames.forEach(processorName -> {
+        // Create a task name for each processor and assign all partitions to each task name.
+        final TaskName taskName = new TaskName(String.format("Task-%s", processorName));
+        groupedMap.put(taskName, ssps);
+      });
 
-    return Collections.singletonMap(taskName, ssps);
+    return groupedMap;
   }
 }
 
 public class AllSspToSingleTaskGrouperFactory implements SystemStreamPartitionGrouperFactory {
+  private static final Logger LOG = LoggerFactory.getLogger(AllSspToSingleTaskGrouperFactory.class);
+
   @Override
   public SystemStreamPartitionGrouper getSystemStreamPartitionGrouper(Config config) {
-    return new AllSspToSingleTaskGrouper(config.getInt(JobConfig.PROCESSOR_ID()));
+    if (!(new TaskConfigJava(config).getBroadcastSystemStreams().isEmpty())) {
+      throw new ConfigException("The job configured with AllSspToSingleTaskGrouper cannot have broadcast streams.");
+    }
+
+    try {
+      String jobCoordinatorFactoryClassName = new JobCoordinatorConfig(config).getJobCoordinatorFactoryClassName();
+      JobCoordinator jc = Util.<JobCoordinatorFactory>getObj(jobCoordinatorFactoryClassName).getJobCoordinator(config);
+      return new AllSspToSingleTaskGrouper(jc.getProcessorNames());
+    } catch (ConfigException ex) {
+      LOG.info("Guessing cluster-based jobCoordinator(Yarn).");
+      final Set<String> processorNames = new HashSet<>();
+      IntStream.range(0,
+          config.getInt(JobConfig.JOB_CONTAINER_COUNT())).forEach(i -> processorNames.add(String.valueOf(i)));
```

Comment:
Why are we still using the constant name here, instead of new JobConfig(config).getContainerCount()?

Reply:
I missed this. Thanks for pointing out.